t/lib/OpenQA/Test/Utils.pm

```diff
@@ -401,9 +401,15 @@ sub setup_share_dir {
     return $sharedir;
 }
 
+sub _test_dir () {
+    return tempdir unless $ENV{OPENQA_FULLSTACK_TEMP_DIR};
+    my $dir = path($ENV{OPENQA_FULLSTACK_TEMP_DIR});
+    return $dir->remove_tree({keep_root => 1});
+}
+
 sub setup_fullstack_temp_dir {
     my ($test_name) = @_;
-    my $tempdir = $ENV{OPENQA_FULLSTACK_TEMP_DIR} ? path($ENV{OPENQA_FULLSTACK_TEMP_DIR}) : tempdir;
+    my $tempdir = _test_dir();
     my $basedir = $tempdir->child($test_name);
```

Comment:
This is the directory which should be deleted.
e.g. full-stack.d is doing:
```
my $tempdir = setup_fullstack_temp_dir('full-stack.d');
```
so it would be `$OPENQA_FULLSTACK_TEMP_DIR/full-stack.d`

Otherwise it could delete a previous dir from another test.